src/libraries/System.Private.Xml/src/System/Xml/Dom/XmlNodeReader.cs

```diff
@@ -1395,6 +1386,7 @@ public override bool MoveToAttribute(string name, string? namespaceURI)
                 return true;
             }
             _readerNav.RollBackMove(ref _curDepth);
+            _nodeType = savedNodeType;
```

Comment:
do `_curDepth` or `_bInReadBinary` need any adjustments? (I've done quick tests with checking `XmlReader.Depth` and that looks consistent across readers so possibly the answer is no)